### PR TITLE
VSP-MS_1

### PR DESCRIPTION
**Pull request by ChaabaneAnas**
## CHNAGES

- Use EXPLAIN ANALYZE on the previous queries to check what is happening.
- Find a way to decrease the execution time of the first query.
- Find a way to improve execution time of the other two queries.
Screenshot from 2022-10-25 15-33-22
Screenshot from 2022-10-25 15-31-17
Screenshot from 2022-10-25 15-34-01
Screenshot from 2022-10-25 15-34-59
Screenshot from 2022-10-25 15-35-33
Screenshot from 2022-10-25 15-35-51

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -91,6 +91,13 @@ insert into visits values (2,12,'aug 3, 2020');
 insert into visits values (3,13,'may 24, 2020');
 insert into visits values (1,13,'jan 11, 2021');
 
+/* Advanced DB Milestone_1*/ 
+
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
+
```
Comment (HAMMAS-SALEEM) [inline data.sql:102]: - [x] This is a pair-programming project means all your team members should contribute in this project. 🤔 But i cannot see any commit from your partner side 👀 Kindly ask your partner to make contributions in this project by adding his commit and resubmit this pull request from your dashboard again for a review 👍

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Review Request 🔴 🚥 🚨

## Hey @ChaabaneAnas :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following GitFlow
:heavy_check_mark: Following some requirements

Good job on implementing some of the requirements, But i cannot proceed review on your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 2]
```diff
@@ -70,3 +70,8 @@ join species s on s.id = species_id
 where v.name = 'Maisy Smith'
 group by specie_name, v.name
 order by count(*) desc limit 1;
+
+/*MileStone 5*/
+CREATE INDEX animal_id_index on visits(animal_id);
+CREATE INDEX vet_id_index on visits(vet_id);
+CREATE INDEX email_id on owners(email);
```
Comment (mukhammadrizooff) [inline queries.sql:77]: - [ ] [OPTIONAL] This is a nice-to-have a suggestion as it won't improve the query that was reported as problematic in the requirements for this project 😃 

As email addresses are not case sensitive, we might need to look for the emails in a case insensitive way. There is an option to reflect that in the index by using:

```
CREATE INDEX ON owners ((lower(email)));
```

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Good implementations ✔️ 
- Descriptive `PR` description and `Commit` history ✔️  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (ChaabaneAnas) [reply queries.sql:77]: Thank you so much mate

**Pull request merged**